ligand/views.py

```diff
@@ -44,26 +112,757 @@ class LigandNameSelection(AbsTargetSelection):
+        return JsonResponse(OK_request_dict)
+
+
+
+class LigandStructuralSearch(TemplateView):
+
+    template_name = 'ligand_structural_search.html'
+    template_name = 'target_details.html'
+
+    def get_context_data(self, **kwargs):
+        cache_key = False
+        mode = 'compact'
+        context = super().get_context_data(**kwargs)
+        param_dict = getLigandStructuralSearchParameters(self.request)
+        search_type = param_dict['search_type']
+        smiles = param_dict['smiles']
+        input_type = param_dict['input_type']
+        with connection.cursor() as cursor:
+            if search_type == 'similarity':
+                similarity_threshold = param_dict['similarity_threshold']
+                cache_key = "ligand_structural_search_" + ",".join([search_type,input_type,smiles,str(similarity_threshold),mode])
+                if not(cache_key != False and cache.has_key(cache_key)):
+                    string_to_replace_by = '^textoreplacebysmiles#'
+                    value = MORGANBV_FP(Value(string_to_replace_by))
+                    q = LigandFingerprint.objects.filter(mfp2__tanimoto=value)
+                    q = q.annotate(sml=TANIMOTO_SML('mfp2', value))
+                    q = q.order_by(TANIMOTO_DIST('mfp2', value))
+                    q = q.values_list('ligand', 'sml')
+                    sql_query = str(q.query).replace('%','%%').replace(string_to_replace_by,'%s')
+                    cursor.execute('SET SESSION rdkit.tanimoto_threshold = %s; ' + sql_query, [similarity_threshold,smiles,smiles,smiles])
+                    cursor_results = cursor.fetchall()
+                    
+
+            elif search_type == 'substructure':
+                if param_dict['stereochemistry']:
+                    stereochemistry = 'true'
+                else:
+                    stereochemistry = 'false'
+                cache_key = "ligand_structural_search_" + ",".join([search_type,smiles,str(stereochemistry),mode])
+                if not(cache_key != False and cache.has_key(cache_key)):
+                    sql_string_placeholder = '^textoreplacebysmiles#'
+                    v = Value(sql_string_placeholder)
+                    if input_type == 'smarts':
+                        m = QMOL(v)
+                    else:
+                        m = MOL(v)
+                    q = LigandMol.objects.filter(molecule__hassubstruct=m).values('ligand')
+                    sql_query = str(q.query).replace(sql_string_placeholder,'%s')
+                    cursor.execute('SET SESSION rdkit.do_chiral_sss=%s; '+sql_query, [stereochemistry,smiles])
+                    cursor_results = cursor.fetchall()
+                    
+            else:
+                raise ValidationError('Unknown ligand structural search parameter: search_type="%s"' % str(search_type))
+            pass
+        # cache.delete(cache_key)
```

Comment:
I guess this was here for testing, please remove

Reply:
Solved in [bb9beb8](https://github.com/protwis/protwis/pull/550/commits/bb9beb8c3231460510e199f66a17a1c512c6c8e0).